File: elixir/6ku/Find The Parity Outlier.ex
# Find The Parity Outlier
# https://www.codewars.com/kata/5526fc09a1bbd946250002dc
defmodule Kata do
  def find_outlier(integers) do
    {x, y} =
      integers
      |> Enum.group_by(&(rem(&1, 2) == 0))
      |> Enum.reject(fn {x, y} -> length(y) > 1 end)
      |> Enum.unzip()

    y |> List.flatten() |> Enum.at(0)
  end
end

File: ruby/6ku/Find The Parity Outlier.rb
# frozen_string_literal: true

# Find The Parity Outlier
# https://www.codewars.com/kata/5526fc09a1bbd946250002dc/train/ruby

def find_outlier(integers)
  integers.partition(&:odd?).find(&:one?).first
end

p find_outlier([1, 2, 3])
